File: Projects/Framework-Samples/Csla/CSharp/PetShop.Silverlight/Source/PetShop.UI.Silverlight/Models/MainPageModel.cs
using System;
using System.Windows;
using System.Windows.Controls;

namespace PetShop.UI.Silverlight
{
    public class MainPageModel : FrameworkElement
    {
        public static readonly DependencyProperty CurrentControlProperty = DependencyProperty.Register("CurrentControl", typeof(UserControl), typeof(MainPageModel), new PropertyMetadata(null));

        public UserControl CurrentControl
        {
            get { return (UserControl)GetValue(CurrentControlProperty); }
            set { SetValue(CurrentControlProperty, value); }
        }

        private static MainPageModel _main;
        public MainPageModel()
        {
            _main = this;
            CurrentControl = new CategoryListPage();
        }

        public static void ShowForm(UserControl form)
        {
            _main.CurrentControl = form;
        }
    }
}

File: trunk/Projects/Framework-Samples/Csla/VisualBasic/PetShop.Silverlight/Source/PetShop.UI.Silverlight/Models/MainPageModel.vb
Imports System.Windows
Imports System.Windows.Controls

Namespace PetShop.UI.Silverlight
    Public Class MainPageModel
        Inherits FrameworkElement
        Public Shared ReadOnly CurrentControlProperty As DependencyProperty = DependencyProperty.Register("CurrentControl", GetType(UserControl), GetType(MainPageModel), New PropertyMetadata(Nothing))

        Public Property CurrentControl() As UserControl
            Get
                Return DirectCast(GetValue(CurrentControlProperty), UserControl)
            End Get
            Set(ByVal value As UserControl)
                SetValue(CurrentControlProperty, value)
            End Set
        End Property

        Private Shared _main As MainPageModel
        Public Sub New()
            _main = Me
            CurrentControl = New CategoryListPage()
        End Sub

        Public Shared Sub ShowForm(ByVal form As UserControl)
            _main.CurrentControl = form
        End Sub
    End Class
End Namespace